### ISSUE TITLE
Preview not updating when you change role

### ISSUE THREAD

**Issue by joebudi** [Budibase/budibase#5777]
This is more of a UX issue.

The template I am using (manufacturing inventory) has a page only available to power users and admins. When I select admin from the 'View as' dropdown and click to view these pages in the preview, it works perfectly. But when I select basic and then select back to admin, the URL does not change. I think this is because BB understands basic is not able to access that URL, and redirects them. 

I am not sure what the solution is - maybe a no-access page when a basic user tries to access the admin page? 

I thought I'd flag this as it caught me out.

https://vimeo.com/707720137/bffe169354

**Comment by aptkingston** [comment]
Yep you're correct - when you try to view an admin page as a basic user it will redirect you to another page. When you change the role back to admin, you're still on that other page, so no change happens.

Basic users don't even know about the existence of admin screens, so we can't show a no-access page when they try to view them. We just treat it as if they hit a URL that doesn't exist and redirect them.

What we could do is add in an edge case that whenever you're in the preview and change the role, maybe that's the only scenario where we can throw up a no-access page. That could also before misleading though because that's not the behaviour that would happen in a real app.
